test-apps/imodel-transformer/src/ElementUtils.ts

```diff
@@ -67,8 +67,22 @@ export namespace ElementUtils {
     }
     if (displayStyle.settings?.excludedElementIds) {
       for (const elementId of displayStyle.settings.excludedElementIds) {
-        iModelDb.elements.getElement(elementId);
+        iModelDb.elements.getElement(elementId); // will throw Error if not a valid Element
       }
     }
   }
+
+  export function queryProvenanceScopeIds(iModelDb: IModelDb): Id64Set {
+    const elementIds = new Set<Id64String>();
+    if (iModelDb.containsClass(ExternalSourceAspect.classFullName)) {
+      const sql = `SELECT Element.Id FROM ${ExternalSourceAspect.classFullName} WHERE Kind=:kind`;
+      iModelDb.withPreparedStatement(sql, (statement: ECSqlStatement) => {

```

Comment:
why not use the async query method instead of `withPreparedStatement`?
```ts
for await (const row of iModelDb.query(sql, {kind: ExternalSourceAspect.Kind.Scope}) {
  const [first] = Object.values(row); // may differ
  elementIds.add(first);
}
```
(it will require queryProvenanceScopeIds to be async)

Reply:
There are currently some restrictions with using `query` that prevent me from using it in the general case. In this specific case, we are talking about 0 to a handful of elements so async won't matter.